src/http/middleware/_jwt.js

```diff
@@ -0,0 +1,26 @@
+
+// This lib is quite large at 339kb is there a better and lighter jwt decoder we could use
+const jsonwebtoken = require('jsonwebtoken')
+
+module.exports = function (arc) {
+  return function jwt (req, res, next) {
+    const route = arc.http.find(route => route[0] === req.method.toLowerCase() && route[1] === req.url)
+    if (route.length > 2) {
+      console.log(req.headers)
+      try {
+        if (!req.headers.authorization) {
+          throw Error('Unauthorized')
+        }
+        // This operation is not secure do not do this in production
+        const tokenValues = jsonwebtoken.decode(req.headers.authorization.split(' ')[1])
+        req.authorizer = { jwt: { claims: tokenValues } }
+      }
+      catch (e) {
+        res.writeHead(401)
+        res.end('{"message":"Unauthorized"}')
+        return
+      }
+    }
+    next()
+  }
+}
```

Comment:
fwiw sending an `authorization` header is a totally normal practice for non-JWT auth. The route being intercepted should first clear a predicate that it's `http[v1]` and configured with the JWT Authorizer before we run any of this.

Reply:
This check only applies if the jwt authorizer is enabled on the endpoint in the arc config. I could probably make that logic a bit better though tbh!